### PR TITLE
Add lecture page

### PR DESCRIPTION
**Pull request by BenLanglois**
Add lectures page with input validation.

Screen Shot 2020-11-29 at 5 34 32 AM
Screen Shot 2020-11-29 at 5 34 27 AM
Screen Shot 2020-11-29 at 5 34 20 AM
Screen Shot 2020-11-29 at 5 34 09 AM

### CODE REVIEW

File: src/utils/YouTube.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  let videoID = null;
+  if (['youtube.com', 'www.youtube.com'].includes(urlObject.hostname)) {
+    // video ID is stored in the 'v' query parameter on the '/watch' page
+    // e.g. https://www.youtube.com/watch?v=VIDEO-ID
+    if (urlObject.pathname === '/watch') {
+      videoID = urlObject.searchParams.get('v');
+    }
+  } else if (urlObject.hostname === 'youtu.be') {
```
Comment (alexgirard) [inline src/utils/YouTube.js:27]:
```suggestion
  } else if (urlObject.hostname === 'youtube') {
```

typo?

File: src/pages/AddLecture.jsx [hunk 2]
```diff
@@ -1,20 +1,206 @@
...
+  border-radius: 10px;
+`;
+
+const FooterContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-end;
+`;
+
+const AddLecture = ({ courseName }) => {
```
Comment (alexgirard) [inline src/pages/AddLecture.jsx:69]: Not important but this entire component might be cleaner as a class component

File: src/utils/YouTube.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    return null;
+  }
+};
+
+/**
+ * See https://stackoverflow.com/questions/2068344/how-do-i-get-a-youtube-video-thumbnail-from-the-youtube-api
+ *
+ * @param {string} id
+ */
+export const getThumbnail = id => `https://img.youtube.com/vi/${id}/sddefault.jpg`;
```
Comment (alexgirard) [inline src/utils/YouTube.js:45]: 💯

File: src/pages/AddLecture.jsx [hunk 4]
```diff
@@ -1,20 +1,206 @@
...
-        theme='gray'
-        width='fit-content'
-        margin='0 auto 1em auto'
-        onClick={() => {}}
-      />
-      <Button text='Hello, world!' theme='violet' onClick={() => {}} />
-    </CenteredPanel>
+    <CenteredContainer>
+      <HeaderContainer>
+        {/* TODO: link to course page */}
```
Comment (alexgirard) [inline src/pages/AddLecture.jsx:171]: We can create a separate component extracted from ariels add course page for this link later :)

**Review by alexgirard — APPROVED** [review]
Looks good minus that one typo,, please rebase!

File: src/utils/YouTube.js (hunk 1 above)
Comment (ariellam) [reply src/utils/YouTube.js:27]: This is the shortened URL I think? e.g. https://youtu.be/ek1I_pcC2cs

**Pull request merged**